lightning/src/ln/channelmanager.rs

```diff
@@ -8285,8 +8285,10 @@ macro_rules! create_offer_builder { ($self: ident, $builder: ty) => {
 		let entropy = &*$self.entropy_source;
 		let secp_ctx = &$self.secp_ctx;
 
-		let path = $self.create_blinded_path_using_absolute_expiry(absolute_expiry)
+		let path = $self.create_blinded_paths_using_absolute_expiry(absolute_expiry)
+			.and_then(|paths| paths.into_iter().next().ok_or(()))
```

Comment:
`DefaultMessageRouter` returns maximum three paths, so it could return too many for an offer. This particular code was just moved from `ChannelManager:: create_blinded_path` since we wanted all of them for potential reply paths.

At some point we'll want to determine how the user specifies they are ok with more than one path.